go-controller/pkg/clustermanager/node/subnet_allocator.go

```diff
@@ -456,3 +474,46 @@ func (snr *subnetAllocatorRange) releaseAllNetworks(owner string) {
 		}
 	}
 }
+
+// listAllNetworks returns all networks in the range
+func (snr *subnetAllocatorRange) listAllNetworks() []*net.IPNet {
```

Comment:
this function reuses logic from https://github.com/ovn-kubernetes/ovn-kubernetes/blob/9ea5b14aa3b2dcf2e6ebf36bbf3330337cf262e1/go-controller/pkg/clustermanager/node/subnet_allocator.go#L368-L428 so I think we should be good here.

Reply:
I guess we could extract this to a common function like `range(func(*net.IPNet) bool)`. We might even want to export it.

Reply:
I extract the common logic in function `listAllNetworks` and `allocateNetwork` to a new function. Not sure if that is what you meant.